transport/handler_server.go

```diff
@@ -253,6 +254,13 @@ func (ht *serverHandlerTransport) writeCommonHeaders(s *Stream) {
 }
 
 func (ht *serverHandlerTransport) Write(s *Stream, hdr []byte, data []byte, opts *Options) error {
+	ht.mu.Lock()
+	done := ht.streamDone
+	ht.mu.Unlock()
```

Comment:
If `WriteStatus` is called after this `unlock`, before `ht.do`, the panic will still happen.

Reply:
Aren't `WriteStatus` calls serialized in gRPC side? `WriteStatus` is called once, and subsequent calls would exit on `ht.streamDone==true`. Not sure how this triggers panics in `WriteStatus`.

Reply:
`WriteStatus` is guaranteed to only execute once, but if this line of code is reached and THEN `WriteStatus` is called and gets all the way to closing `ht.writes` before the `ht.do()` here, then we would end up with the same panic.

Should `WriteStatus` call `ht.Close` before closing `ht.writes`?  I think that would also fix the problem.  Then the code inside `do` will avoid writing to the closed channel.